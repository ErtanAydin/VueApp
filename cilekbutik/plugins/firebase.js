import * as firebase from 'firebase/app'
import 'firebase/auth'

const firebaseConfig = {
    apiKey: "AIzaSyC7kInVEj6L35XB8NX_iwdgKvpnzpkK_WU",
    authDomain: "cilekbutik-9efd9.firebaseapp.com",
    projectId: "cilekbutik-9efd9",
    storageBucket: "cilekbutik-9efd9.appspot.com",
    messagingSenderId: "128564693192",
    appId: "1:128564693192:web:34362c3303bb2a162b3dfd",
    measurementId: "G-V1W8HH0S1K"
  };

  let app = null;
  if(!firebase.apps.length){
      app = firebase.initializeApp(firebaseConfig);
  }

export default firebase;