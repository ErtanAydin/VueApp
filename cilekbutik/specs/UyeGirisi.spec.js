import { mount, RouterLinkStub } from "@vue/test-utils";
import UyeGirisi from "@/components/UyeGirisi";
import * as firebase from 'firebase'

const onAuthStateChanged = jest.fn()

const getRedirectResult = jest.fn(() => {
  return Promise.resolve({
    user: {
      email:'a@hotmail.com',
      sifre: '123456'
    }
  })
})


const initializeApp = jest
  .spyOn(firebase, 'initializeApp')
  .mockImplementation(() => {
    return {
      auth: () => {
        return {
          getRedirectResult
        }
      }
    }
  })

  jest.spyOn(firebase, 'auth').mockImplementation(() => {
    return {
      onAuthStateChanged,
      currentUser: {
        email: 'a@hotmail.com',
        sifre: '123456'
      },
      getRedirectResult,

    }
  })

describe("methods", () => {
  test("onayla", async () => {
    const mockRouter = {
      push: jest.fn()
    }
    const wrapper = mount(UyeGirisi, {
      props: {
        authenticated: true
      },
      global: {
        mocks: {
          $router: mockRouter
        }
      },
      stubs: {
        NuxtLink: RouterLinkStub,
      }});
      wrapper.vm.email = 'a@hotmail.com'
      wrapper.vm.sifre = '123456'
      expect(firebase.auth).toHaveBeenCalled()
      expect(wrapper.vm.email).toBe('a@hotmail.com')
      expect(wrapper.vm.sifre).toBe('123456')
      expect(mockRouter.push)
      expect(console.clear())
      expect(console.log(wrapper.vm.user))
      expect(console.log('a@hotmail.com' + ' giris yapildi') )
  })
})

describe("mounted with firebase", () => {
  test("mounted with firebase test", async () => {
    const wrapper = mount(UyeGirisi, {
      stubs: {
        NuxtLink: RouterLinkStub
      }});
      
      expect(firebase.auth).toHaveBeenCalled()
  })
})

describe("UyeGirisi", () => {
  test("mounts properly", () => {
    const wrapper = UyeGirisi;
    expect(wrapper).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = mount(UyeGirisi, {
      stubs: {
        NuxtLink: RouterLinkStub
      }});
    expect(wrapper.html()).toMatchSnapshot();
  });
});

describe ('data', () => {
  test('data test', () => {
    const wrapper = mount(UyeGirisi,{
    stubs: {
      NuxtLink: RouterLinkStub
    }});
    expect(wrapper.vm.user).toBe('')
    expect(wrapper.vm.eposta).toBe('')
    expect(wrapper.vm.sifre).toBe('')
    expect(wrapper.vm.errors).toBe('')
  })
})