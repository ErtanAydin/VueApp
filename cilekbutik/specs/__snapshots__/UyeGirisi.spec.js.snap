// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UyeGirisi renders properly 1`] = `
<div class="UyeGirisiDiv">
  <h3 class="UyeGirisiBaslık">
    Üye Girişi
  </h3>
  <form>
    <p class="UyeGirisiEposta"> E-Posta Adresiniz</p> <input type="text" class="UyeGirisiInput">
    <p class="UyeGirisiSifre"> Şifre</p> <input type="password" class="UyeGirisiInput">
    <div class="SagText"><a href="https://www.cilekbutik.net/uye-giris?sayfa=sifre" class="UyeGirisiLink">Şifremi Hatırlat</a></div> <button type="submit" class="GirisYapButon"> Giriş Yap </button>
  </form>
</div>
`;
